### DISCUSSION TITLE
need help with quartz.server 3.5.0 database configuration, net core 6 and SQL Server 2017

### DISCUSSION THREAD

**Discussion by glocken** [quartznet/quartznet#1855] [General] [answered]
I'm trying to connect Quartz.Server 3.5 to SQL Server 2017 but the service doesn't start when using JobStoreTX.
When I use RAMJobStore quartz service starts with no problem, but when replaced with JobStoreTX it doesn't start.

Here are some workarounds that fixed the issue for others but didn't work for me:

- Unblocking zip file.  [explained here](https://github.com/quartznet/quartznet/issues/205)
- Assigned "Full Control" permission to the main folder in case of raising other permission related stuff.
- Move "Quartz Server" folder to "Program Files (x86)".
- Changing provider to SqlServer-20
- Adding this line to config --> quartz.jobStore.lockHandler.type = Quartz.Impl.AdoJobStore.UpdateLockRowSemaphore, Quartz

I've used the older version of quartz.server before, but with this version I didn't have much luck.

This is my quartz.config database config:

quartz.jobStore.type = Quartz.Impl.AdoJobStore.JobStoreTX, Quartz
quartz.jobStore.tablePrefix = dbo.QRTZ_
quartz.jobStore.driverDelegateType = Quartz.Impl.AdoJobStore.SqlServerDelegate, Quartz
quartz.jobStore.dataSource = myDs
quartz.dataSource.myDS.connectionString = Server=localhost\sql2017;Database=---;User Id=---;Password=---
quartz.dataSource.myDS.provider = SqlServer
quartz.jobStore.clustered = false
quartz.jobStore.useProperties = true

**Comment by lahma** [comment]
Have tried enabling logging to get some logs?

**Reply by glocken** [reply]
Yes, the log file is being created but it's empty.

this is my log4net.config content:
```
<log4net>
	<appender name="LogFileAppender" type="log4net.Appender.RollingFileAppender" >
		<param name="File" value="MyQuartzLog.txt" />
		<param name="AppendToFile" value="true" />
		<rollingStyle value="Size" />
		<maxSizeRollBackups value="10" />
		<maximumFileSize value="10MB" />
		<staticLogFileName value="true" />
		<layout type="log4net.Layout.PatternLayout">
			<param name="ConversionPattern" value="%-5p%d{yyyy-MM-dd hh:mm:ss} – %m%n" />
		</layout>
	</appender>
	<root>
		<level value="DEBUG" />
		<appender-ref ref="LogFileAppender" />
	</root>
</log4net>
```

**Comment by glocken** [comment]
I was able to find the exception by running the dll using `dotnet quartz.server.dll` and got this:

System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Data.SqlClient, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.
File name: 'Microsoft.Data.SqlClient, Culture=neutral, PublicKeyToken=null'

I added  `"Microsoft.Data.SqlClient": "5.0.1"` to Quartz.Server.deps.config.json but didn't work.

How can I define this dependency in quartz.server?

**Reply by lahma** [reply]
Via csproj PackageReference.

**Comment by glocken** [comment, marked as answer]
Thank you @lahma,
I republished Quartz.Server project with missing nuget package and it worked.

Using `dotnet quartz.server.dll` without 'install' or 'start' helped alot.
